backend/src/govsg/services/govsg-callback.service.ts

```diff
@@ -133,9 +142,9 @@ const parseTemplateMessageWebhook = async (
     })
     // throwing error here to return 400
     // this is because callbacks could hit this endpoint before the messageId is updated in GovsgOp table
-    // only do this for sending as this is unlikely to happen for other statuses
+    // only do this for sent and delivered as this is unlikely to happen for other statuses
     // this will trigger a retry from WhatsApp, which will hit this endpoint again after messageId is updated
-    if (whatsappStatus === WhatsAppMessageStatus.sent) {
+    if (statusesWhichRequireMessageId.has(whatsappStatus)) {
```

Comment:
Am I correct to say that this change is required because its possible that whatsapp directly sends a "delivered" status update? i.e, there's no guarantee that we will receive "sent" first before "delivered" for a particular messageId?

Reply:
Yes. Your statements are correct. Here's [one of the docs](https://developers.facebook.com/docs/whatsapp/conversation-types#message-status) which support this.